code/test/tokens.js

```diff
@@ -0,0 +1,472 @@
+  })
+
+  // ERC20 Token Testing (Transfer, Mint, Track, SpendingLimit)
+  it('Wallet_CommitReveal: ERC20(Transfer, Mint, Track) must commit and reveal successfully', async () => {
+    // Create Wallets and tokens
+    const alice = await makeWallet(1, accounts[1])
+    const bob = await makeWallet(2, accounts[2])
+    const { testerc20 } = await makeTokens(alice.lastResortAddress)
+    let aliceBalanceERC20
+    let aliceWalletBalanceERC20
+    let bobBalanceERC20
+    let bobWalletBalanceERC20
+    // transfer ERC20 tokens from alice.lastResortAddress (which owns the tokens) to alices wallet
+    await testerc20.transfer(alice.wallet.address, 1000, { from: alice.lastResortAddress })
+    aliceWalletBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    assert.equal(1000, aliceWalletBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet succesful')
+    // alice transfers tokens to bob
+    await tokenTransfer(
+      {
+        wallet: alice,
+        operationType: ONEConstants.OperationType.TRANSFER_TOKEN,
+        tokenType: ONEConstants.TokenType.ERC20,
+        contractAddress: testerc20.address,
+        dest: bob.wallet.address,
+        amount: 100
+      }
+    )
+    // OneWallet Items that have changed
+    // check alice and bobs balance
+    aliceBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    aliceWalletBalanceERC20 = await alice.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    bobBalanceERC20 = await testerc20.balanceOf(bob.wallet.address)
+    bobWalletBalanceERC20 = await bob.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    assert.equal(900, aliceBalanceERC20, 'Transfer of 100 ERC20 tokens from alice.wallet succesful')
+    assert.equal(900, aliceWalletBalanceERC20, 'Transfer of 100 ERC20 tokens from alice.wallet succesful and wallet balance updated')
+    assert.equal(100, bobBalanceERC20, 'Transfer of 100 ERC20 tokens to bob.wallet succesful')
+    assert.equal(100, bobWalletBalanceERC20, 'Transfer of 100 ERC20 tokens to bob.wallet succesful and wallet balance updated')
+    // Alice Items that have changed - nonce, lastOperationTime, Commits, trackedTokens
+    assert.notStrictEqual(await alice.wallet.nonce, alice.state.nonce, 'alice wallet.nonce should have been changed')
+    // alice.state.nonce = parseInt(alice.state.nonce) + 1
+    alice.state.nonce = await alice.wallet.getNonce()
+    // lastOperationTime
+    assert.notStrictEqual(await alice.wallet.lastOperationTime(), alice.state.lastOperationTime, 'alice wallet.lastOperationTime should have been updated')
+    alice.state.lastOperationTime = await alice.wallet.lastOperationTime()
+    // commits
+    assert.notDeepEqual(await alice.wallet.getAllCommits(), alice.state.allCommits, 'alice wallet.allCommits should have been updated')
+    alice.state.allCommits = await alice.wallet.getAllCommits()
+    // tracked tokens
```

Comment:
It should check whether this particular ERC20 token is now tracked by Alice, but not Bob

Reply:
Agreed - will implement.